packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceModal.tsx

```diff
@@ -0,0 +1,483 @@
+import { useProject } from '../../../hooks/useProject';
+import { useProjectAccess } from '../../../hooks/useProjectAccess';
+import {
+    AccessDescription,
+    AccessName,
+    AccessRole,
+    AccessSelectSubtitle,
+    AccessSelectTitle,
+    AccessWrapper,
+    AddUsersWrapper,
+    ChangeAccessButton,
+    DialogFooter,
+    FlexWrapper,
+    MemberAccess,
+    OpenShareModal,
+    ShareButton,
+    ShareTag,
+    UserName,
+    UserRole,
+    UserTag,
+    YouLabel,
+} from './ShareSpaceModal.style';
+
+import { MenuItem2 } from '@blueprintjs/popover2';
+import {
+    useAddSpaceShareMutation,
+    useDeleteSpaceShareMutation,
+    useUpdateMutation,
+} from '../../../hooks/useSpaces';
+import { useApp } from '../../../providers/AppProvider';
+export interface ShareSpaceProps {
+    space: Space;
+    projectUuid: string;
+}
+
+export interface AccessOption {
+    title: string;
+    description?: string;
+    selectDescription: string;
+    value: string;
+}
+const StyledSpinner = () => <Spinner size={16} style={{ margin: 12 }} />;
+
+const capitalize = (word: string): string =>
+    word ? `${word.charAt(0).toUpperCase()}${word.slice(1).toLowerCase()}` : '';
+
+const enum SpaceAccessType {
+    PRIVATE = 'private',
+    PUBLIC = 'public',
+}
+const SpaceAccessOptions: AccessOption[] = [
+    {
+        title: 'Restricted access',
+        description: 'Only invited members can access',
+        selectDescription: 'Only invited members can access',
+        value: SpaceAccessType.PRIVATE,
+    },
+    {
+        title: 'Full access',
+        description: 'All project members can access',
+        selectDescription:
+            'All project members can access with their project permissions',
+        value: SpaceAccessType.PUBLIC,
+    },
+];
+
+const enum UserAccessAction {
+    KEEP = 'keep',
+    DELETE = 'delete',
+}
+
+const UserAccessOptions: AccessOption[] = [
+    {
+        title: 'viewer',
+        selectDescription: `This permission has been inherited from user's project access`,
+        value: UserAccessAction.KEEP,
+    },
+    {
+        title: 'No access',
+        selectDescription: `Remove user's access`,
+        value: UserAccessAction.DELETE,
+    },
+];
+
+const renderAccess: ItemRenderer<AccessOption> = (
+    access,
+    { handleClick, handleFocus, modifiers, query },
+) => {
+    if (!modifiers.matchesPredicate) {
+        return null;
+    }
+    return (
+        <MenuItem2
+            multiline={true}
+            active={modifiers.active}
+            disabled={modifiers.disabled}
+            key={access.value}
+            onClick={handleClick}
+            onFocus={handleFocus}
+            text={
+                <>
+                    <AccessSelectTitle>{access.title}</AccessSelectTitle>
+                    <AccessSelectSubtitle>
+                        {access.selectDescription}
+                    </AccessSelectSubtitle>
+                </>
```

Comment:
second element should have a very small margin or not have it at all
<img width="533" alt="CleanShot 2022-11-16 at 13 17 58@2x" src="https://user-images.githubusercontent.com/962095/202140599-2b8dc997-e2c2-42c9-a44b-e16fe77ed988.png">